server.js

```diff
@@ -433,6 +441,23 @@ export class FilesCollection extends FilesCollectionCore {
       return;
     }
     WebApp.connectHandlers.use((httpReq, httpResp, next) => {
+      if (!this.disableCORS && !!~httpReq._parsedUrl.path.indexOf(`${this.downloadRoute}/`) && !httpResp.headersSent) {
+        if (originRE.test(httpReq.headers.origin)) {
+          httpResp.setHeader('Access-Control-Allow-Credentials', 'true');
+          httpResp.setHeader('Access-Control-Allow-Origin', httpReq.headers.origin);
+        }
+
+        if (httpReq.method === 'OPTIONS') {
+          httpResp.setHeader('Access-Control-Allow-Methods', 'GET, POST, OPTIONS');
+          httpResp.setHeader('Access-Control-Allow-Headers', 'Range');
```

Comment:
@s-ol I've tested the update. I like the solution in general, but uploads are failing without proper value of `Access-Control-Allow-Headers`. How about following suggestion:
`allowedAccessControlHeaders` can be moved to the top next to the origin port regex.
```suggestion
          const allowedAccessControlHeaders = ['range', 'content-type', 'x-mtok', 'x-start', 'x-chunkid', 'x-fileid', 'x-eof']
          const accessControlRequestHeaders = httpReq.headers['access-control-request-headers'] || ''
          const accessControlAllowedHeaders = accessControlRequestHeaders
            .split(',')
            .filter(header => allowedAccessControlHeaders.includes(header))
            .join(',') 
          httpResp.setHeader('Access-Control-Allow-Headers', accessControlAllowedHeaders);
```

Reply:
interesting, i haven't experienced that issue but i will try this suggestion tomorrow :)

Reply:
@s-ol I'm surprised on your experience as well. 
Have you checked request/response headers in dev tools for your case? 
Are you using protected uploads?

FYI, I've tested this again on Android 8.0 and iOS 12. 
Android sends OPTIONS request first, iOS sends POST request directly (without OPTIONS).
In both cases headers `content-type, x-mtok, x-start` are set. And CORS Policy rejects the response without those headers in `Access-Control-Allow-Headers`

Reply:
> Are you using protected uploads?

both `config.protected` and `config.onBeforeUpload` are functions that validate `this.userId` in our setup.

ahh... i just found this in our UI code 😅:

            transport: Meteor.isCordova ? 'ddp' : 'http',

I guess you have been testing HTTP uploads properly? I will test native uploads and merge your changes soon 👍

Reply:
Yeah, I use HTTP for uploads on native devices. 
Actually I'd like to create a very simple example project for files uploads/downloads in Cordova and add it to the wiki as I found it quite challenging to make it work. What do you think on this?

Reply:
@vbelolapotkov do you mean including the changes to Meteor-Files as per this branch? What have you been needing to configure outside of this branch (and the corresponding ostrio:cookies branch) specifically for Cordova?

In general it sounds like a good idea. I am going to resolve this thread since the headers have been added.

Reply:
@s-ol no any other changes for this branch.  

Some challenges I had and which I'd like to share as part of the example project are following:

- how to properly read a file from device (Android/iOS) to upload it with Meteor-Files. Mostly Cordova specific stuff.
- how to download a file with Meteor-Files by sending x-mtok as a url param. Actually this was required for images because of missing xmtok cookie on localhost.
- how to save/open file on a native device. Mostly Cordova stuff as well but it could demonstrate how to integrate Meteor-Files package into the app.

Any feedback is welcome